Update api versions

siiruli opened the pull request:
Closes #12

siiruli committed:
Refactor: organize ui folder to be more practical

- Move main views to their own folder
- Move list items together. They already share code and will
even more in the future.
- Refactor the itinerary code to be a bit more logical. There
is now a form, a query, and a result.
- move form_components to components.

siiruli committed:
Refactor query results

Create a new GraphQLQueryResult widget that handles the possible errors
in query results. Use that instead of copy-pasted builder functions in
itinerary and stops.

Rename StopListWidget to StopQueryResults so that Results widgets now
consistently mean widgets that fetch and display the results of a graphql
query.

siiruli committed:
Update API versions

Update API endpoints to new versions.
Remove some query parameters that no longer exist.
Add a note to README about using graphql_codegen.

The plan() query used in itinerary_query is deprecated,
so it should be replaced with planConnection().

https://github.com/siiruli/bussit-flutter/issues/12

ollpu commented on lib/ui/views/stop_search_view.dart:
```diff
@@ -40,7 +40,6 @@ class _SearchStopsState extends State<SearchStops> {
           Expanded(
               child: StopQueryResults(
             searchName: searchName,
-            maxResults: 10,
```
Bad things seem to happen without this limit (making the name field empty freezes the application)

ollpu commented on lib/ui/widgets/itinerary/itinerary_form.dart:
```diff
@@ -55,7 +55,6 @@ class _ItineraryFormState extends State<ItineraryForm>
           nResults: 12,
           time: formData.datetime,
           arriveBy: formData.arriveBy,
-          allowBikeRental: _allowBikeRental,
```
Please remove the definition of `_allowBikeRental`, it isn't used anywhere else.

ollpu commented on lib/ui/widgets/query_results/stop_results.dart:
```diff
@@ -22,29 +23,26 @@ class StopListWidget extends HookWidget {
-  }
+Widget stopListBuilder(Map<String, dynamic> result) {
+  final data = Query$StopData.fromJson(result);
+  List<dynamic> stops = (List<dynamic>.empty() +
+          (data.stations ?? List<dynamic>.empty()) +
+          (data.stops ?? List<dynamic>.empty()))
```
```suggestion
  List<dynamic> stops = ((data.stations ?? List<dynamic>.empty()) +
          (data.stops ?? List<dynamic>.empty()))
```

siiruli committed:
remove allowBikerental

siiruli committed:
Try to fix responsiveness of stop search

The stop search widget became unresponsive.
Limit the number of stops shown,
fetch results only when user presses enter,
and don't show results for less than 3 search
characters.

siiruli committed:
Add comment

ollpu committed:
Disable cache for stop queries
